jax/lax/lax.py

```diff
@@ -2356,6 +2356,7 @@ def _broadcast_batch_rule(batched_args, batch_dims, sizes):
 
 def _broadcast_in_dim_impl(operand, shape, broadcast_dimensions):
   if type(operand) is xla.DeviceArray:
+    shape = _broadcast_in_dim_shape_rule(operand, shape, broadcast_dimensions)
```

Comment:
The check looks good, but if its purpose is to find usage errors, shouldn't it be closer to the API, e.g. lax.broadcast_in_dim?